### DISCUSSION TITLE
Group by open/closed in GitHub project insights

### DISCUSSION THREAD

**Discussion by widal001** [community/community#78307] [Projects and Issues]
### Select Topic Area

Product Feedback

### Feature Area

Projects

### Request

Stacked bar charts in GitHub projects currently support grouping by single select fields. And historical charts currently group by open/closed. It would be great to have the option to group values by open/closed when the x-axis is something _other_ than time.

### Context

Our stakeholders often want to know the % completion across a number of different categories (e.g. sprint, milestone, etc.) and the closest that we can come to showing this is grouping by status. However, if our engineering team wants to have 3+ statuses (e.g. "Todo", "In progress", "In review", "Done") it becomes a bit harder to simply visualize done vs open within the chart.

There are some workarounds to this -- for example, creating a single select field with just two values "open" and "closed" -- but it becomes cumbersome to maintain parity between that field and the actual closed/open status of the ticket.

<img width="1080" alt="Screenshot 2023-12-08 at 4 59 51 PM" src="https://github.com/community/community/assets/21350331/9d77a6c4-3be8-4eb3-80a3-436eed6fc9bc">

### How it should work

Ideally this new feature would let you simply select "Closed status" or some other name to indicate open vs closed from the "Group by (optional)" configuration settings on a chart.

<img width="405" alt="Screenshot 2023-12-08 at 5 00 58 PM" src="https://github.com/community/community/assets/21350331/72596eaf-1426-4ba1-80c2-38f300365c7b">

### Alternatives considered

We've explored a few alternatives, none of which have been a perfect substitute:

- Grouping by status. Works okay, but can result in cluttered charts that don't quite meet stakeholder needs.
- Custom single select field with just two options (e.g. closed, open). Not ideal because it requires manually keeping this in sync with the issue's closed vs open status.
- Exporting the data and visualizing it outside of GitHub insights. Not ideal because it requires a separate infrastructure to run and calculate these details.

### Guidelines

- [X] I have read the above statement and can confirm my post is relevant to the GitHub feature areas [Issues](https://docs.github.com/en/issues/tracking-your-work-with-issues/about-issues) and/or [Projects](https://docs.github.com/en/issues/planning-and-tracking-with-projects/learning-about-projects/about-projects).

**Comment by slinna** [comment]
@widal001 if a 3rd party app is an option, I suggest taking a look at the [Analytics & Reports](https://github.com/marketplace/analytics-reports) add-on in the Marketplace. It provides [advanced charts](https://screenful.com/tour/charts) for GitHub repos and projects. You can [learn more](https://screenful.com/github) or give it a try by starting the trial via the marketplace.

**Reply by widal001** [reply]
We were hoping to have this feature directly within GitHub insights, but this is a helpful interim option. Thanks for sharing this tool!
